src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -393,7 +452,317 @@ public virtual bool TryInsertExpansion(int startPositionInSubjectBuffer, int end
+                            new XElement(snippetNamespace + "Title", new XText(methodName)),
+                            new XElement(snippetNamespace + "Author", "Microsoft"),
+                            new XElement(snippetNamespace + "Description"),
+                            new XElement(snippetNamespace + "HelpUrl"),
+                            new XElement(snippetNamespace + "Shortcut", methodName)),
+                        new XElement(
+                            snippetNamespace + "Snippet",
+                            new XElement(snippetNamespace + "Declarations", declarations.ToArray()),
+                            new XElement(
+                                snippetNamespace + "Code",
+                                new XAttribute(snippetNamespace + "Language", "csharp"),
+                                new XCData(template.ToString()))))));
+        }
+
+        private void OnModelUpdated(object sender, Model e)
+        {
+            AssertIsForeground();
+
+            if (e is null)
+            {
+                return;
+            }
+
+            if (!_state.IsFullMethodCallSnippet)
+            {
+                return;
+            }
+
+            var document = SubjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+            if (document is null)
+            {
+                return;
+            }
+
+            var newSymbolKey = (e.SelectedItem as AbstractSignatureHelpProvider.SymbolKeySignatureHelpItem)?.SymbolKey ?? default;
+            var newSymbol = newSymbolKey.Resolve(document.Project.GetRequiredCompilationAsync(CancellationToken.None).WaitAndGetResult(CancellationToken.None), cancellationToken: CancellationToken.None).GetAnySymbol();
+            if (newSymbol is not IMethodSymbol method)
+                return;
+
+            MoveToSymbol(_state._symbols, method, CancellationToken.None);
+        }
+
+        private async Task<ImmutableArray<ISymbol>> GetSymbolsAsync(
+            SnapshotPoint caretPosition,
+            CancellationToken cancellationToken)
+        {
+            var document = SubjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+            if (document is null)
+            {
+                return ImmutableArray<ISymbol>.Empty;
+            }
+
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var token = await semanticModel.SyntaxTree.GetTouchingTokenAsync(caretPosition.Position, cancellationToken).ConfigureAwait(false);
+            var semanticInfo = semanticModel.GetSemanticInfo(token, document.Project.Solution.Workspace, cancellationToken);
+            return semanticInfo.ReferencedSymbols;
+        }
+
+        public void MoveToSymbol(ImmutableArray<IMethodSymbol> symbols, IMethodSymbol symbol, CancellationToken cancellationToken)
+        {
+            AssertIsForeground();
+
+            if (ExpansionSession is null)
```

Comment:
doing this check here seems... strange.  i woudl have expected it in the caller (which was doing the rest of the checks).  i would expect this method to only be really checking hte methods to decide if it can proceed or not.

Reply:
➡️ The caller does the checks that relate to Signature Help. This method does the checks that relate to snippets.